src/libraries/System.Net.Http/src/System/Net/Http/HttpRequestMessage.cs

```diff
@@ -201,32 +200,10 @@ private void InitializeValues(HttpMethod method, Uri? requestUri)
 
         internal bool MarkAsSent()
         {
-            return Interlocked.CompareExchange(ref _sendStatus, MessageAlreadySent, MessageNotYetSent) == MessageNotYetSent;
+            return Interlocked.Exchange(ref _sendStatus, MessageAlreadySent) == MessageNotYetSent;
```

Comment:
Was it necessary change?

Reply:
This line change comes from reverting some changes from #40338. As we moved the instrumentation into HttpClient, we don't need the extra synchronization logic around the Request message anymore.